examples/EmailerAgent/emailer/agent.py

```diff
@@ -0,0 +1,416 @@
+        self.smtp_address = self.config.get("smtp-address", None)
+        self.from_address = self.config.get("from-address", None)
+        self.to_address = self.config.get("to-addresses", None)
+        self.smtp_port = self.config.get("smtp-port", None)
+        self.smtp_username = self.config.get("smtp-username", None)
+        self.smtp_password = self.config.get("smtp-password", None)
+        self.smtp_tls = self.config.get("smtp-tls")
+        self.allow_frequency_minutes = self.config.get("allow-frequency-minutes", 60)
+        self._allow_frequency_seconds = self.allow_frequency_minutes * 60
+        self.smtp_tls = self.config.get("smtp-tls",None)
+        self.default_config = dict(smtp_address=self.smtp_address,
+                                   from_address=self.from_address,
+                                   to_addresses=self.to_address,
+                                   smtp_port=self.smtp_port,
+                                   smtp_username=self.smtp_username,
+                                   smtp_password=self.smtp_password,
+                                   smtp_tls = self.smtp_tls,
+                                   allow_frequency_minutes=self.allow_frequency_minutes,
+                                   alert_from_address=self.from_address,
+                                   alert_to_addresses=self.to_address,
+                                   send_alerts_enabled=True,
+                                   record_sent_emails=True)
+        self.current_config = None
+        self.vip.config.set_default("config", self.default_config)
+
+        self.vip.config.subscribe(self.configure_main,
+                                  actions=["NEW", "UPDATE"], pattern="*")
+
+        # Keep track of keys that have been added to send with.
+        self.tosend = {}
+
+        def onstart(sender, **kwargs):
+            self.vip.pubsub.subscribe('pubsub', topics.PLATFORM_SEND_EMAIL,
+                                      self.on_email_message)
+
+            self.vip.pubsub.subscribe('pubsub', topics.ALERTS_BASE,
+                                      self.on_alert_message)
+            self.vip.pubsub.subscribe('pubsub',
+                                      prefix=topics.ALERTS.format(agent_class='',
+                                                                  agent_uuid=''),
+                                      callback=self.on_alert_message)
+
+        self.core.onstart.connect(onstart, self)
+        self.sent_alert_emails = defaultdict(int)
+
+#    @Core.periodic(10)
+#    def accessService(self):
+#       headers = {
+#           "something": 'something'
+           #"to-addresses": ['alpha.beta@foo.com', 'bob-and-joe@bar.com']
+#       }
+
+#       message = {
+#           "subject": "Something is not right",
+#           "message": "Somethign is down"
+#       }
+
+       self.vip.pubsub.publish('pubsub', topic='platform/send_email',headers=headers ,message=message)
```

Comment:
Also I do believe that this line would cause an error because headers and message aren't defined in this block